v2/bundle/spiffebundle/bundle.go

```diff
@@ -3,175 +3,411 @@ package spiffebundle
 import (
 	"crypto"
 	"crypto/x509"
+	"encoding/json"
 	"io"
+	"io/ioutil"
+	"sync"
 	"time"
 
 	"github.com/spiffe/go-spiffe/v2/bundle/jwtbundle"
 	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
+	"github.com/spiffe/go-spiffe/v2/internal/x509util"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/zeebo/errs"
+	"gopkg.in/square/go-jose.v2"
 )
 
+const (
+	x509SVIDUse = "x509-svid"
+	jwtSVIDUse  = "jwt-svid"
+)
+
+var (
+	spiffebundleErr = errs.Class("spiffebundle")
+)
+
+type bundleDoc struct {
+	jose.JSONWebKeySet
+	SequenceNumber uint64 `json:"spiffe_sequence,omitempty"`
+	RefreshHint    int64  `json:"spiffe_refresh_hint,omitempty"`
```

Comment:
These should  be pointers so we have explicit behavior on set v.s. unset values. For example, as written, somebody couldn't set a sequence number of zero and have it show up in the marshaled bundle.
```suggestion
	SequenceNumber *uint64 `json:"spiffe_sequence,omitempty"`
	RefreshHint    *int64  `json:"spiffe_refresh_hint,omitempty"`
```